crates/wasmtime/src/instance.rs

```diff
@@ -450,6 +450,38 @@ impl Instance {
         Some(item)
     }
 
+    /// Looks up exported [`Extern`] value by a [`ModuleExport`] value.
+    ///
+    /// This is similar to [`Instance::get_export`] but uses a [`ModuleExport`] value to avoid
+    /// string lookups where possible.
+    ///
+    /// This method will search the module for an export with a matching entity index and return
+    /// the value, if found.
+    ///
+    /// Returns `None` if there was no export with a matching entity index.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `store` does not own this instance.
+    pub fn get_module_export(
+        &self,
+        mut store: impl AsContextMut,
+        export: &ModuleExport,
+    ) -> Option<Extern> {
+        let store = store.as_context_mut().0;
```

Comment:
Could the above method delegate to this one? Additionally could this one lookup in `data.exports` to use the cache if it's already filled in? And finally, mind adding a link to `Module::get_export_index` in the docs of this function?

Reply:
Also, can this method perform a check that `export.module` is the same as the module used to create this instance? That prevents a mistake where you get a `ModuleExport` from one module but look it up in the wrong instance.